lib/metadataTypes/Folder.js

```diff
@@ -258,39 +258,38 @@ class Folder extends MetadataType {
      * @param {Object} metadata metadata of the folder
      * @returns {Promise} Promise
      */
-    static create(metadata) {
+    static async create(metadata) {
+        if (metadata.Parent && metadata.Parent.ID && metadata.Parent.ID === 0) {
+            Util.logger.error(
+                `${this.definition.type}-${metadata.ContentType}.create:: Cannot create Root Folder: ${metadata.Name}`
+            );
+            return {};
+        }
         const tempCustomerKey = metadata.CustomerKey;
```

Comment:
maybe these need moving into the try catch block?

Reply:
what for? dont see how it could cause an error

Reply:
if they arent used outside, they should be inside.

Reply:
path is used in the catch part.
but I removed tempCustomerKey - from my test just now it does not seem to be necessary anymore